### PR TITLE
Go 1.14.4 and dependencies upgrade

### PR DESCRIPTION
**Pull request by sylr**
## Changes

- Change golang version to latest available
- `go get -u ./...`

To be merged right after the release of v0.13.0 ?

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -35,7 +35,7 @@ We use *breaking* word for marking changes that are not backward compatible (rel
 
 ### Changed
 
-- [#2194](https://github.com/thanos-io/thanos/pull/2194) Updated to golang v1.14.2
+- [#2194](https://github.com/thanos-io/thanos/pull/2194) Updated to golang v1.14.3
```
Comment (bwplotka) [inline CHANGELOG.md:38]: Are you sure this makes sense?

File: go.mod [hunk 2]
```diff
@@ -1,68 +1,115 @@
...
+	gopkg.in/ini.v1 v1.56.0 // indirect
+	gopkg.in/yaml.v2 v2.3.0
+	gopkg.in/yaml.v3 v3.0.0-20200506231410-2ff61e1afc86
+	honnef.co/go/tools v0.0.1-2020.1.4 // indirect
+	howett.net/plist v0.0.0-20200419221736-3b63eb3a43b5 // indirect
+	k8s.io/api v0.18.2 // indirect
+	k8s.io/apimachinery v0.18.2 // indirect
+	k8s.io/client-go v1.5.1 // indirect
+	k8s.io/utils v0.0.0-20200520001619-278ece378a50 // indirect
+	sigs.k8s.io/yaml v1.2.0 // indirect
```
Comment (bwplotka) [inline go.mod:112]: we need `go mod tidy`

File: CHANGELOG.md (hunk 1 above)
Comment (sylr) [reply CHANGELOG.md]: 🤦

File: go.mod (hunk 2 above)
Comment (sylr) [reply go.mod:112]: It already has been. AFAIU those `// indirect` are due to modules that have no `go.mod` files.

**Comment by sylr** [comment]
@bwplotka could you do the Maintainer's bit ? :)

**Pull request closed without merging**